OpenEdXMobile/src/main/java/org/edx/mobile/module/analytics/AnalyticsProvider.java

```diff
@@ -0,0 +1,327 @@
+package org.edx.mobile.module.analytics;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.google.inject.Singleton;
+
+import org.edx.mobile.util.images.ShareUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An analytics provider class whose function is to contain all the analytics services we are using
+ * in our app, and call their respective APIs to send out analytics events.
```

Comment:
> containing all the providers in it and spit out events using those providers

This describes a "delegator" or "registry", not a "provider". As you mention, that would better describe the various `Analytics` implementations that are registered here, instead.